### PR TITLE
[Perf] Accumulate Conformer attention scores with baddbmm

### PR DESCRIPTION
**Pull request by Levius-Fubuki**
## Purpose

Conformer relative attention currently materializes the content scores, adds
the shifted positional scores, and then launches a separate in-place scale.
This PR uses standard `torch.baddbmm` to accumulate the scaled content scores
directly into the scaled positional scores:

```text
before: {bmm: 2, add: 1, mul_: 1}
after:  {bmm: 1, baddbmm: 1}
```

This removes one score-sized intermediate and the separate add/scale kernels.
It keeps one portable PyTorch path with no custom kernel, architecture gate,
device branch, or device-specific tuning.

### Regression test

```bash
python -m pytest -q tests/models/multimodal/test_conformer_encoder.py
```

The test instantiates the real module and compares final output and post-softmax
attention against the pre-change implementation for sequence lengths 2 and 8.
It covers masking and production-style batch-1 positional broadcasting.

### Lint and pre-commit

```bash
pre-commit run --files \
  tests/models/multimodal/test_conformer_encoder.py \
  vllm/model_executor/models/conformer_encoder.py
```

### RTX 4090 CUDA Graph microbenchmark

```bash
python tools/bench_vllm_conformer_baddbmm.py \
  --mode benchmark \
  --batches 1 4 16 --heads 4 8 \
  --sequences 64 128 256 512 1024 \
  --head-dims 32 64 --dtypes float16 bfloat16 \
  --warmups 25 --iterations 200 --repeats 5 \
  --output portability-benchmark.csv
```

Hardware/software: NVIDIA RTX 4090, driver 595.71.05, CUDA 13.0,
PyTorch 2.13.0+cu130. No configuration changes.

| B | H | T | D | dtype | Old p10/p50/p90 (us) | New p10/p50/p90 (us) | p50 speedup | Old/New extra peak allocation (MiB) |
|---:|---:|---:|---:|:---|:---|:---|---:|---:|
| 1 | 4 | 64 | 64 | FP16 | 9.18 / 9.18 / 9.20 | 8.28 / 8.28 / 8.29 | 1.11x | 0.16 / 0.12 |
| 1 | 8 | 128 | 32 | BF16 | 9.62 / 9.80 / 10.54 | 8.71 / 9.55 / 9.56 | 1.03x | 1.25 / 1.00 |
| 4 | 8 | 512 | 32 | FP16 | 170.92 / 170.94 / 171.09 | 138.66 / 138.72 / 139.07 | 1.23x | 80.03 / 64.03 |
| 16 | 4 | 256 | 64 | BF16 | 100.56 / 100.62 / 100.73 | 93.93 / 94.03 / 94.05 | 1.07x | 40.03 / 32.03 |
| 16 | 8 | 512 | 64 | FP16 | 1082.85 / 1083.32 / 1083.42 | 955.74 / 955.93 / 956.04 | 1.13x | 320.13 / 256.13 |
| 16 | 8 | 1024 | 64 | BF16 | 4084.58 / 4085.03 / 4085.41 | 3514.84 / 3515.28 / 3515.57 | 1.16x | 1280.25 / 1024.25 |

Across all 120 CUDA Graph rows, the geometric-mean speedup was approximately
1.11x; the median was 1.09x and the observed minimum was 1.03x. All 240
eager/CUDA Graph rows passed numerical validation with zero OOM rows and zero
harness-reported allocation regressions.

The non-contiguous Q/K operands require the same two reshape copies in both the
old `matmul` path and the new path. The measured allocation reduction comes
from removing the additional score-sized add result. FP16/BF16 changed-input
CUDA Graph replay and a focused `torch.compile(fullgraph=True)` smoke check also
passed.

The standalone harness, raw CSV, and gate output are available in the
[benchmark artifact Gist](https://gist.github.com/Levius-Fubuki/c125b9c6696331d65acef27d3d7fe5f1).

## Test Result

```text
2 passed
```

`ruff-format`, `ruff-check`, SPDX-header, forbidden-import, new-`torch.cuda`
API checks, and `git diff --check` passed locally.

## Limitations

These are isolated score-chain results from one RTX 4090 run, not end-to-end
Conformer throughput. No performance claim is made for other architectures or
backends. Full upstream cross-backend CI still requires a maintainer `ready` or
`verified` label. The positional tensor uses batch `B` in the performance
harness; the real-module regression test separately covers production batch-1
broadcasting.

## AI assistance

AI assistance was used during implementation, testing, benchmarking, and PR
drafting. The human contributor reviewed and takes responsibility for the
change and evidence.

### CODE REVIEW

**Comment by DarkLight1337** [comment]
cc @AllenDou

**Comment by AllenDou** [comment]
Looks good to me. I also tested this patch(FireRedASR2), and it works well. So, if all the test cases are green, I think this PR can be merged.

File: tests/models/multimodal/test_conformer_encoder.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (DarkLight1337) [inline tests/models/multimodal/test_conformer_encoder.py:1]: Move this into `tests/models/multimodal`

Comment (Levius-Fubuki) [reply tests/models/multimodal/test_conformer_encoder.py:1]: Moved this to `tests/models/multimodal/test_conformer_encoder.py`. The focused test (`2 passed`) and relevant pre-commit hooks pass from the new location.

**Review by DarkLight1337 — APPROVED** [review]
Thanks for optimizing

**Comment by Levius-Fubuki** [comment]
The two failures appear unrelated to this PR. Both are isolated to `DeepseekV4ForConditionalGeneration`:

- The MI300 job fails `test_registry_imports[DeepseekV4ForConditionalGeneration]` because `is_text_generation_model(...)` returns `False` (`1 failed, 396 passed, 22 skipped`).
- H200 shard 7 fails while initializing the DeepSeek V4 engine (`1 failed, 27 passed`); the aggregate H200 check is reported as successful.

This PR only changes the Conformer relative-position attention path, and the remaining relevant model, multimodal, and compilation checks passed. The full logs are available in [Buildkite #87055](https://buildkite.com/vllm/ci/builds/87055/list).

Could you please retry the remaining failed MI300 CI job?

**Comment by DarkLight1337** [comment]
Force merging
